gusto/diagnostics.py

```diff
@@ -401,6 +401,42 @@ def name(self):
         return self.field1+"_perturbation"
 
 
+class Precipitation(DiagnosticField):
+    name = "Precipitation"
+
+    def setup(self, state):
+        if not self._initialised:
+            space = state.spaces("DG1", state.mesh, "DG", 1)
+            super(Precipitation, self).setup(state, space=space)
+
+            rain = state.fields('rain')
+            rho = state.fields('rho')
+            v = state.fields('rainfall_velocity')
+            phi = TestFunction(space)
+            flux = TrialFunction(space)
+            n = FacetNormal(state.mesh)
+            un = 0.5 * (dot(v, n) + abs(dot(v, n)))
+            self.flux = Function(space)
+            self.total_water = Function(space)
+
+            if space.extruded:
+                self.ds = ds_b + ds_t + ds_v
+            else:
+                self.ds = ds
```

Comment:
I think this was done because otherwise you are redefining `firedrake.ds` which gives a lint error, and assigning it to be a member seemed a convenient enough way to do that.

Reply:
OK, yes, I see (I hate python's scoping rules).

Yeah, either use a different variable name or else maybe write:

```python
            a = phi * flux * dx
            L = phi * rain * un * rho
            if space.extruded:
                L = L*(ds_b + ds_t + ds_v)
            else:
                L = L*ds
```

Or similar?